invenio_curations/services/components.py

```diff
@@ -87,6 +90,38 @@ def _check_update_request(
                 system_identity, request["id"], request, uow=self.uow
             )
 
+    def _prepare_data(self, data, current_draft):
+        """Build and return fresh new maps only with fields that should be compared later in the process.
+
+        Method used in context of comment processing.
+        """
+        supported_fields = ["metadata", "custom_fields"]
+        new_data = {}
+        new_crt_draft = {}
+        for field in supported_fields:
+            new_data[field] = data[field]
+            new_crt_draft[field] = current_draft[field]
+
+        return new_data, new_crt_draft
+
+    def _get_diff_processor(self):
+        """Construct and return diff processor."""
+        return DiffProcessor(
+            configured_elements=current_curations_service.comments_mapping,
+            comment_template_file=current_curations_service.comment_template_file,
+        )
```

Comment:
is this intended to be reused elsewhere?
otherwise, i think creating a wrapper for a single function/constructor call might be a bit overzealous with the clean code approach

creating too many small functions creates a *lot* of mental overhead when trying to follow the logic flow